packages/vuelidate/src/core.js

```diff
@@ -134,7 +138,8 @@ function createAsyncResult (rule, model, $pending, $dirty) {
           $pending.value = !!$pendingCounter.value
           $invalid.value = true
         })
-    }
+    },
+    { flush: 'sync' }
```

Comment:
This was necessary otherwise it calls it on next tick, thus the `$validate` method fails to trigger properly. I am not sure of the complications of this, so I might just await a tick in `$validate` instead?

Reply:
Good question. It shouldn’t require a synced update to work properly. What exactly failed without it?

Reply:
Well, `$validate` triggers dirty, and then checks what has errors, before it resolves. But watchers are async, and flush after at tick. Which means for `$validate` to work, we would need to await those to finish, and then check for error states.